rotkehlchen/chain/base/modules/basenames/decoder.py

```diff
@@ -0,0 +1,222 @@
+            self,
+            context: DecoderContext,
+            label_hash: str,
+            node: bytes | None = None,
+    ) -> str | None:
+        """Try to get the name corresponding to a labelhash.
+        Checks the cache first and then queries the base.org api.
+        Returns the name or None if unable to retrieve a name.
+        """
+        label_hash = label_hash.removeprefix('0x')
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (found_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.ENS_LABELHASH, label_hash),
+            )):
+                return found_name
+
+        try:
+            response_data = requests.get(
+                url=f'{BASENAMES_API}/{int(label_hash, 16)}',
+                timeout=CachedSettings().get_timeout_tuple(),
+            ).json()
+        except (requests.exceptions.RequestException, JSONDecodeError, ValueError) as e:
+            log.error(
+                f'Failed to retrieve name for labelhash {label_hash} '
+                f'from Basenames api due to {e!s}',
+            )
+            return None
+
+        try:
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                globaldb_set_unique_cache_value(
+                    write_cursor=write_cursor,
+                    key_parts=(CacheType.ENS_LABELHASH, label_hash),
+                    value=(name := response_data['name']),
+                )
+        except (TypeError, KeyError):
+            log.error(
+                f'Failed to retrieve name for labelhash {label_hash} '
+                f'due to unexpected response from Basenames api: {response_data}',
+            )
+            return None
+
+        return name
+
+    def _decode_registrar_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != NAME_REGISTERED_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        try:
+            _, decoded_data = decode_event_data_abi_str(
+                tx_log=context.tx_log,
+                abi_json=NAME_REGISTERED_ABI,
+            )
+        except DeserializationError as e:
+            log.debug(f'Failed to decode Basename registered event due to {e!s}')
+            return DEFAULT_DECODING_OUTPUT
+
+        fullname = f'{decoded_data[0]}.base.eth'
+        expires = decoded_data[1]
+        spend_event = receive_event = None
+        for event in context.decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.asset == A_ETH and event.address == context.tx_log.address:  # noqa: E501
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.counterparty = CPT_BASENAMES
+                event.notes = f'Register Basenames name {fullname} for {event.balance.amount} ETH until {self.timestamp_to_date(expires)}'  # noqa: E501
```

Comment:
```suggestion
                event.notes = f'Register {self.display_name} name {fullname} for {event.balance.amount} ETH until {self.timestamp_to_date(expires)}'  # noqa: E501
```
might as well utilize the attribute.